lib/daru/maths/statistics/vector.rb

```diff
@@ -70,24 +66,99 @@ def factors
           reject_values(*Daru::MISSING_VALUES).uniq.reset_index!
         end
 
-        # Maximum element of the vector.
+        # Returns the maximum value present in the vector.
         #
-        # @param return_type [Symbol] Data type of the returned value. Defaults
-        #   to returning only the maximum number but passing *:vector* will return
-        #   a Daru::Vector with the index of the corresponding maximum value.
-        def max return_type=:stored_type
-          max_value = @data.max
-          if return_type == :vector
-            Daru::Vector.new({index_of(max_value) => max_value}, name: @name, dtype: @dtype)
-          else
-            max_value
-          end
+        # @example
+        #
+        #   dv = Daru::Vector.new (["Tyrion", "Daenerys", "Jon Starkgaryen"]), index: Daru::Index.new([:t, :d, :j])
+        #   #=>
+        #   #   #<Daru::Vector(3)>
+        #   #       t   Tyrion
+        #   #       d   Daenerys
+        #   #       j   Jon Starkgaryen
+        #
+        #   dv.max
+        #   #=> "Tyrion"
+        #
+        #   dv.max { |i| i.size }
+        #   #=> "Jon Starkgaryen"
+        def max(&block)
+          arr = block_given? ? @data.data.map(&block) : @data.data
+          @data.data[arr.index(arr.max)]
+        rescue
+          @data.to_a.max
```

Comment:
Since a `Daru::Accessors::ArrayWrapper` class (which `@data` is an instance of) already has  a method called `max`.  (https://github.com/SciRuby/daru/blob/master/lib/daru/accessors/array_wrapper.rb#L89)

Also, why the `rescue`?